### DISCUSSION TITLE
Can the same dialog be reused in different views?

### DISCUSSION THREAD

**Discussion by nir34** [AvaloniaUtils/DialogHost.Avalonia#46] [Q&A] [answered]
For example, if I define a dialogHost in viewA, can I also open this dialogHost in viewB without having to redefine the same dialogHost in viewB？I tried defining the style of the dialog in DialogHost.DataTemplates or Resources, but it seems that neither approach allows the dialog to be used by other views.

**Comment by SKProCH** [comment]
Hello, @nir34 

You should use different DialogHosts. But yes, you can use the same dialog content in multiple views as long as it isn't a control itself (viewmodel for example).

Controls as dialog content can't be used in multiple views simultaneously, since control can't have more than one parent.

**Reply by nir34** [reply]
Thank you for your response. So, if I've created a DialogHost, like:
`<DialogHost Identifier="xx">
    <dialogHostAvalonia:DialogHost.DataTemplates>
        <DataTemplate x:DataType="vm">
            <Border></Border>
        </DataTemplate>
    </dialogHostAvalonia:DialogHost.DataTemplates>
</DialogHost>
`


And I want to be able to pop up this dialog in different views, do I have to copy the above code into each view?

**Reply by SKProCH** [reply]
Probably yes.

But you can actually extract DataTemplates and put it, for example, into App.axaml to avoid repeating

**Reply by nir34** [reply]
DataTemplates cannot specify x:Key, so it seems I cannot reference it elsewhere. If I define a DataTemplate with a key in app.axaml, and DialogHost.DataTemplates does not seem to have a parameter that can accept this key, am I missing something?

**Reply by SKProCH** [reply]
Probably there is no such parameter currently. Can you link some documentation about this, so I can implement it?

**Reply by nir34** [reply, marked as answer]
This can solve the problem：

var template = Application.Current.FindResource("xxx");
diaogHost.DataTemplates.Add(template as IDataTemplate);
